v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -0,0 +1,142 @@
+			return extensions.PostReconcileCheckResultSuccess(), nil
+		}
+
+		// call sync api only when protection status is ProtectionError and error code is usererror
+		if protectionStatus == "ProtectionError" {
+			protectionStatusErrorCode := ""
+			protectionStatusErrorCode = strings.ToLower(*backupInstance.Status.Properties.ProtectionStatus.ErrorDetails.Code)
+			log.V(Debug).Info(fmt.Sprintf("########################## Protection Error code is  %q ##########################", protectionStatusErrorCode))
+
+			if protectionStatusErrorCode != "" && strings.Contains(protectionStatusErrorCode, "usererror") {
+				id, _ := genruntime.GetAndParseResourceID(backupInstance)
+				subscription := id.SubscriptionID
+				rg := id.ResourceGroupName
+				vaultName := id.Parent.Name
+
+				log.V(Debug).Info("########################## Starting NewBackupInstancesClient for Backup Instance ##########################")
+				clientFactory, err := armdataprotection.NewClientFactory(subscription, armClient.Creds(), armClient.ClientOptions())
+				if err != nil {
+					log.Error(err, "failed to create armdataprotection client")
+				}
+
+				var parameters armdataprotection.SyncBackupInstanceRequest
+				parameters.SyncType = to.Ptr(armdataprotection.SyncTypeDefault)
+				log.V(Debug).Info("########################## Starting BeginSyncBackupInstance for Backup Instance ##########################")
+				poller, err := clientFactory.NewBackupInstancesClient().BeginSyncBackupInstance(ctx, rg, vaultName, backupInstance.AzureName(), parameters, nil)
+				if err != nil {
+					log.Error(err, "failed BeginSyncBackupInstance to get the result")
+				}
+
+				_, err = poller.PollUntilDone(ctx, nil)
+				if err != nil {
+					log.Error(err, "failed BeginSyncBackupInstance to poll the result")
+				}
+			}
+		}
+	}
+	return extensions.PostReconcileCheckResultFailure("Backup Instance is in non terminal state"), nil
+}
+
+func (ext *BackupVaultsBackupInstanceExtension) PreReconcileCheck(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	owner genruntime.MetaObject,
+	_ *resolver.Resolver,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+	_ extensions.PreReconcileCheckFunc,
+) (extensions.PreReconcileCheckResult, error) {
+	log.V(Debug).Info("########################## Starting Pre-reconcilation for Backup Instance ##########################")
+	backupInstance, ok := obj.(*dataprotection.BackupVaultsBackupInstance)
+	if !ok {
+		return extensions.PreReconcileCheckResult{},
+			errors.Errorf("cannot run on unknown resource type %T, expected *dataprotection.BackupVaultsBackupInstance", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = backupInstance
+
+	log.V(Debug).Info("########################## Ending Pre-reconcilation for Backup Instance ##########################")
+
+	return extensions.ProceedWithReconcile(), nil
+}
```

Comment:
Your implementation is "do nothing" - either add the missing functionality or remove it entirely.

Reply:
(In fact, by not calling the passed instance of `extensions.PreReconcileCheckFunc`, this implementation isn't "do nothing", it is "break ASO".)

Reply:
NO, If I remove this Post reconciliation flow is not getting called. I have seen that behaviour in ASO and that why I am keeping it explicitly

Reply:
I have verified it mutiple times to check on this in my local testing

Reply:
sure, please check once

Reply:
Thanks, I am get this removed